### PR TITLE
Refactor tables in the AppView

### PR DESCRIPTION
**Pull request by andresmgot**
Ref: #848

Refactor common code used by the different tables in the AppView: WorkloadTable, ServiceTable, SecretsTable and OtherResourcesTable.

Now everything is a `ResourceTable` and depending on the type of the resource (`kind`) it renders a different type of row.

### CODE REVIEW

File: dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        case "Deployment":
+          return <DeploymentItemRow resource={resource.item} />;
+        case "StatefulSet":
+          return <StatefulSetItemRow resource={resource.item} />;
+        case "DaemonSet":
+          return <DaemonSetItemRow resource={resource.item} />;
+        case "Service":
+          return <ServiceItem resource={resource.item} />;
+        case "Secret":
+          // TODO: Set double type
```
Comment (andresmgot) [inline dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.tsx]: I need to change the type here

File: dashboard/src/components/AppView/ResourceTable/ResourceTable.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { SecretColumns } from "./ResourceItem/SecretItem/SecretItem";
+import { ServiceColumns } from "./ResourceItem/ServiceItem/ServiceItem";
+import { StatefulSetColumns } from "./ResourceItem/StatefulSetItem/StatefulSetItem";
+
+interface IWorkloadItemTableProps {
+  title: string;
+  resourceRefs: ResourceRef[];
+}
+
+class WorkloadTable extends React.Component<IWorkloadItemTableProps> {
```
Comment (andresmgot) [inline dashboard/src/components/AppView/ResourceTable/ResourceTable.tsx]: `ResourceTable`

File: dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          const i = (resource.item as any) as ISecret;
```
Comment (prydonius) [inline dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.tsx]: hmm maybe SecretItem should just accept IResource and internally change it to ISecret?

Comment (andresmgot) [reply dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.tsx]: I would see it weird for the `SecretItem` to accept a `IResource` when it only works for an `ISecret`. It's better to do the conversion before. It's actually the generic "TableItem" the one that can receive different types of resources.

Comment (prydonius) [reply dashboard/src/components/AppView/ResourceTable/ResourceItem/ResourceTableItem.tsx]: You're right, I was just thinking this given that ServiceItem and other resources look like they accept IResources

**Pull request merged**